sdk/samples/iot/paho_iot_hub_pnp_sample.c

```diff
@@ -66,954 +49,917 @@ static az_span boot_time_span;
-  remainder = az_span_copy(remainder, mqtt_url_suffix);
-  az_span_copy_u8(remainder, null_terminator);
-
-  return AZ_OK;
+  // Publish the twin document request.
+  mqtt_publish_message(twin_document_topic_buffer, AZ_SPAN_NULL, 0);
 }
 
-// Read the user environment variables used to connect to IoT Hub
-static az_result read_configuration_and_init_client(void)
+static void receive_messages(void)
 {
-  az_span cert = AZ_SPAN_FROM_BUFFER(x509_cert_pem_file);
-  AZ_RETURN_IF_FAILED(
-      read_configuration_entry(ENV_DEVICE_X509_CERT_PEM_FILE, NULL, false, cert, &cert));
-
-  az_span trusted = AZ_SPAN_FROM_BUFFER(x509_trust_pem_file);
-  AZ_RETURN_IF_FAILED(
-      read_configuration_entry(ENV_DEVICE_X509_TRUST_PEM_FILE, "", false, trusted, &trusted));
+  int rc;
+  char* topic = NULL;
+  int topic_len = 0;
+  MQTTClient_message* message = NULL;
 
-  az_span device_id_span = AZ_SPAN_FROM_BUFFER(device_id);
-  AZ_RETURN_IF_FAILED(
-      read_configuration_entry(ENV_DEVICE_ID, NULL, false, device_id_span, &device_id_span));
+  // Continue to receive commands or device twin messages while device is operational
+  while (device_operational)
+  {
+    LOG(" "); // Formatting.
+    LOG("Waiting for Command or Device Twin message.\n");
 
-  az_span iot_hub_hostname_span = AZ_SPAN_FROM_BUFFER(iot_hub_hostname);
-  AZ_RETURN_IF_FAILED(read_configuration_entry(
-      ENV_IOT_HUB_HOSTNAME, NULL, false, iot_hub_hostname_span, &iot_hub_hostname_span));
+    if (((rc
+          = MQTTClient_receive(mqtt_client, &topic, &topic_len, &message, TIMEOUT_MQTT_RECEIVE_MS))
+         != MQTTCLIENT_SUCCESS)
+        && (rc != MQTTCLIENT_TOPICNAME_TRUNCATED))
+    {
+      LOG_ERROR("Failed to receive message: MQTTClient return code %d.", rc);
+      exit(rc);
+    }
+    else if (message == NULL)
+    {
+      LOG_ERROR("Receive message timeout expired: MQTTClient return code %d.", rc);
+      exit(rc);
+    }
+    else if (rc == MQTTCLIENT_TOPICNAME_TRUNCATED)
+    {
+      topic_len = (int)strlen(topic);
+    }
+    LOG_SUCCESS("Client received message from the service.");
 
-  // Paho requires that the MQTT endpoint be of the form ssl://<HUB ENDPOINT>:8883
-  AZ_RETURN_IF_FAILED(
-      create_mqtt_endpoint(mqtt_endpoint, (int32_t)sizeof(mqtt_endpoint), iot_hub_hostname_span));
+    on_message_received(topic, topic_len, message);
 
-  // Initialize the hub client with the hub host endpoint and the default connection options
-  az_iot_hub_client_options options = az_iot_hub_client_options_default();
-  options.model_id = model_id;
-  AZ_RETURN_IF_FAILED(az_iot_hub_client_init(
-      &client,
-      az_span_slice(iot_hub_hostname_span, 0, (int32_t)strlen(iot_hub_hostname)),
-      az_span_slice(device_id_span, 0, (int32_t)strlen(device_id)),
-      &options));
+    MQTTClient_freeMessage(&message);
+    MQTTClient_free(topic);
 
-  return AZ_OK;
+    // Send a telemetry message
+    send_telemetry_message();
```

Comment:
as discussed this should be sending messages in between waiting for messages. say every 8-10 seconds.

Reply:
Please check the new commit. Thanks.